azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1002,57 +1050,63 @@ private void cleanOldExecutionLogs(long millis) {
       Pair<String, String>... params) throws IOException {
-    URIBuilder builder = new URIBuilder();
-    builder.setScheme("http").setHost(host).setPort(port).setPath("/executor");
-
-    builder.setParameter(ConnectorParams.ACTION_PARAM, action);
+    List<Pair<String, String>> paramList = new ArrayList<Pair<String,String>>();
 
-    if (executionId != null) {
-      builder.setParameter(ConnectorParams.EXECID_PARAM,
-          String.valueOf(executionId));
+    // if params = null
+    if(params != null) {
+      paramList.addAll(Arrays.asList(params));
     }
 
-    if (user != null) {
-      builder.setParameter(ConnectorParams.USER_PARAM, user);
-    }
-
-    if (params != null) {
-      for (Pair<String, String> pair : params) {
-        builder.setParameter(pair.getFirst(), pair.getSecond());
-      }
-    }
+    paramList
+      .add(new Pair<String, String>(ConnectorParams.ACTION_PARAM, action));
+    paramList.add(new Pair<String, String>(ConnectorParams.EXECID_PARAM, String
+      .valueOf(executionId)));
+    paramList.add(new Pair<String, String>(ConnectorParams.USER_PARAM, user));
 
-    URI uri = null;
-    try {
-      uri = builder.build();
-    } catch (URISyntaxException e) {
-      throw new IOException(e);
-    }
+    Map<String, Object> jsonResponse =
+      callExecutorForJsonObject(host, port, "/executor", paramList);
 
-    ResponseHandler<String> responseHandler = new BasicResponseHandler();
+    return jsonResponse;
+  }
 
-    HttpClient httpclient = new DefaultHttpClient();
-    HttpGet httpget = new HttpGet(uri);
-    String response = null;
-    try {
-      response = httpclient.execute(httpget, responseHandler);
-    } catch (IOException e) {
-      throw e;
-    } finally {
-      httpclient.getConnectionManager().shutdown();
-    }
+  /*
+   * Helper method used by ExecutorManager to call executor and return json
+   * object map
+   */
+  private Map<String, Object> callExecutorForJsonObject(String host, int port,
+    String path, List<Pair<String, String>> paramList) throws IOException {
+    String responseString =
+      callExecutorForJsonString(host, port, path, paramList);
 
     @SuppressWarnings("unchecked")
     Map<String, Object> jsonResponse =
-        (Map<String, Object>) JSONUtils.parseJSONFromString(response);
+      (Map<String, Object>) JSONUtils.parseJSONFromString(responseString);
     String error = (String) jsonResponse.get(ConnectorParams.RESPONSE_ERROR);
     if (error != null) {
       throw new IOException(error);
     }
-
     return jsonResponse;
   }
 
+  /*
+   * Helper method used by ExecutorManager to call executor and return raw json
+   * string
+   */
+  private String callExecutorForJsonString(String host, int port, String path,
+    List<Pair<String, String>> paramList) throws IOException {
+    if (paramList == null) {
+      paramList = new ArrayList<Pair<String, String>>();
+    }
+
+    ExecutorApiClient apiclient = ExecutorApiClient.getInstance();
```

Comment:
seems no cache a copy of the reference here, you may just call ExecutorApiClient.getInstance().httpGet()